From 350318916004cc6f828e76de6889df07394caa3a Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Sat, 1 Dec 2018 20:30:16 -0500
Subject: [PATCH 22/45] gl: window-gbm: Remove unneeded extra function

The cleanup function was only called in _close() which was only
calling that function. This indirection didn't make much sense.

(cherry picked from commit 6f7fbbfa71914fe99830c716b0e8254bb9047cfe)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c | 27 +++++++++------------------
 1 file changed, 9 insertions(+), 18 deletions(-)

diff --git a/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c b/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
index b0eb258d63..1678fa28a4 100644
--- a/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
+++ b/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
@@ -50,8 +50,6 @@ static void gst_gl_window_gbm_egl_close (GstGLWindow * window);
 static void gst_gl_window_gbm_egl_draw (GstGLWindow * window);
 
 static gboolean gst_gl_window_gbm_init_surface (GstGLWindowGBMEGL * window_egl);
-static void gst_gl_window_gbm_egl_cleanup (GstGLWindowGBMEGL * window_egl);
-
 
 
 static void
@@ -112,7 +110,15 @@ gst_gl_window_gbm_egl_close (GstGLWindow * window)
 {
   GstGLWindowGBMEGL *window_egl = GST_GL_WINDOW_GBM_EGL (window);
 
-  gst_gl_window_gbm_egl_cleanup (window_egl);
+  if (window_egl->gbm_surf != NULL) {
+    if (window_egl->current_bo != NULL) {
+      gbm_surface_release_buffer (window_egl->gbm_surf, window_egl->current_bo);
+      window_egl->current_bo = NULL;
+    }
+
+    gbm_surface_destroy (window_egl->gbm_surf);
+    window_egl->gbm_surf = NULL;
+  }
 
   GST_GL_WINDOW_CLASS (gst_gl_window_gbm_egl_parent_class)->close (window);
 }
@@ -344,21 +350,6 @@ cleanup:
 }
 
 
-static void
-gst_gl_window_gbm_egl_cleanup (GstGLWindowGBMEGL * window_egl)
-{
-  if (window_egl->gbm_surf != NULL) {
-    if (window_egl->current_bo != NULL) {
-      gbm_surface_release_buffer (window_egl->gbm_surf, window_egl->current_bo);
-      window_egl->current_bo = NULL;
-    }
-
-    gbm_surface_destroy (window_egl->gbm_surf);
-    window_egl->gbm_surf = NULL;
-  }
-}
-
-
 /* Must be called in the gl thread */
 GstGLWindowGBMEGL *
 gst_gl_window_gbm_egl_new (GstGLDisplay * display)
-- 
2.11.0

